unlock-app/src/services/web3Service.js

```diff
@@ -451,13 +451,15 @@ export default class Web3Service extends EventEmitter {
   /**
    * This function will trigger events based on smart contract events
    * @private
-   * @param {*} name
-   * @param {*} params
+   * @param {string} transactionHash
+   * @param {string} contractAddress
+   * @param {string} name
+   * @param {object} params
    */
-  emitContractEvent(transaction, name, params) {
+  emitContractEvent(transactionHash, contractAddress, name, params) {
```

Comment:
This was the crux of the bug :/
We used to pass a transaction object all around but this is now only the hash of it!